### PR TITLE
[#33] Set up Axios and Redux

### PR DESCRIPTION
**Pull request by mjordonez2001**
### Issues
- #33 

### What?
- Configures axios
- Makes a `.env.sample` file for API key
- Configures Redux
- Configures Redux toolkit

### CODE REVIEW

**Review by olets — COMMENTED** [review]
- Let's update the readme to have a step of duplicating the sample env file
- Does the build tooling automatically find and use dotenv? Or is there something missing here?

**Comment by solomonhawk** [comment]
We caught up about this yesterday @mjordonez2001, but I figured I'd just add the comments here too:

I don't think you need the `dotenv` package unless you're using `.env` values _outside_ of the app (outside of the files that vite is loading/transforming, such as within the `vite.config.ts` file itself) which requires calling `dotenv.config()` somewhere.

I'd expect to see an instance of axios set up (using `axios.create(..)`) with some base options/configuration and ready to be used by the app.

In general I think your instincts for making granular changes is great - it makes reviewing code a breeze. It's a little hard to draw a line in the sand for what set of changes is _too_ small to warrant its own PR/review but this is bordering on that in my opinion. I think you could incorporate a few more changes before asking for a review and merge.

File: src/utils/axios.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import axios from "axios";
+
+const ACCESS_TOKEN: string = import.meta.env.VITE_API_ACCESS_TOKEN;
```
Comment (olets) [inline src/utils/axios.ts:3]: import.meta.env is cool. Is that Vite built-in dotenv-like functionality? new js 🔮 ?

Comment (solomonhawk) [reply src/utils/axios.ts:3]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import.meta
https://vitejs.dev/guide/env-and-mode.html

Pretty cool

**Review by solomonhawk — APPROVED** [review]
This is fine to merge if you want to tackle the redux/rtk-query stuff separately. The PR title indicates to me that redux is meant to be part of this change but the PR description doesn't, is it?

File: src/utils/axios.ts (hunk 1 above)
Comment (olets) [reply src/utils/axios.ts:3]: > console.log(import.meta); // { url: "file:///home/user/my-module.js" }

neato, that'd be a great way to something like `data-the-file-to-look-at-if-you-want-to-edit-this="…"` in the final

**Pull request merged**